src/reconcile/rec_visibility.c

```diff
@@ -454,13 +455,6 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
     if (has_newer_updates)
         r->leave_dirty = true;
 
-    /*
-     * We should restore the update chains to the new disk image if there are newer updates in
-     * eviction.
-     */
-    if (has_newer_updates && F_ISSET(r, WT_REC_EVICT))
-        r->cache_write_restore = true;
```

Comment:
If we reverted the key promote change don't we need to keep this?

Reply:
I moved it into  if (__rec_need_save_upd)

The previous version I think is not correct for in memory and fixed length column store.

Reply:
Cool